lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py

```diff
@@ -46,8 +46,20 @@
     kind:
       description:
         - Type of credential being added.  The ssh choice refers to a Tower Machine credential.
-      required: True
+      required: False
       choices: ["ssh", "vault", "net", "scm", "aws", "vmware", "satellite6", "cloudforms", "gce", "azure_rm", "openstack", "rhv", "insights", "tower"]
+    credential_type:
```

Comment:
Can now specify custom credential type using `credential_type` field

Reply:
I could have sworn I remembered some pattern for requiring one out of a list of fields, but can't find it now.